### PR TITLE
Base Miner and Data Augmentation Updates

### PR DESCRIPTION
**Pull request by dylanuys**
- Added option to turn on data augmentation for test data, aligning model evaluation with validator data
    - Also made setting setting train, val, test transforms cleaner
- Improved image quality resulting from data augmentation pipeline
    - Removed extra resize that would happen if randomresizecrop was used in addition to base transforms
    - Using bilinear interpolation for rotates
- will update base weights in subsequent PR

### CODE REVIEW

File: bitmind/image_transforms.py [hunk 1]
```diff
@@ -80,18 +82,20 @@ def __call__(self, img):
...
     transforms.Lambda(lambda t: t.expand(3, -1, -1) if t.shape[0] == 1 else t),
 ])
 
-# example data augmentation
-random_image_transforms = ComposeWithParams([
-    RandomResizedCropWithParams(256, scale=(0.2, 1.0), ratio=(1.0, 1.0)),
+# data augmentation
+random_aug_transforms = ComposeWithParams([
+    transforms.ToTensor(),
+    transforms.Lambda(lambda t: t.expand(3, -1, -1) if t.shape[0] == 1 else t),
```
Comment (dylanuys) [inline bitmind/image_transforms.py]: used to concatenate these onto the base transforms, now they're kept independent

File: bitmind/constants.py [hunk 2]
```diff
@@ -42,6 +42,8 @@
     ]
 }
 
+TARGET_IMAGE_SIZE = (256, 256)
```
Comment (benliang99) [inline bitmind/constants.py:45]: Question (non-blocking): Are we using "Target" to describe all images output by our validator?

File: bitmind/image_transforms.py [hunk 3]
```diff
@@ -80,18 +82,20 @@ def __call__(self, img):
...
     transforms.Lambda(lambda t: t.expand(3, -1, -1) if t.shape[0] == 1 else t),
 ])
 
-# example data augmentation
+# data augmentation
 random_aug_transforms = ComposeWithParams([
-    RandomRotationWithParams(20),
-    RandomResizedCropWithParams(256, scale=(0.2, 1.0), ratio=(1.0, 1.0)),
+    transforms.ToTensor(),
+    transforms.Lambda(lambda t: t.expand(3, -1, -1) if t.shape[0] == 1 else t),
```
Comment (aliang322) [inline bitmind/image_transforms.py]: can leave comment about using this for image type catching/handling (convert grayscale/1 channel images to 3 channel)

Comment (aliang322) [reply bitmind/image_transforms.py]: Should we also handle images with 4 channels (alpha channel?)

File: base_miner/train_detector.py [hunk 4]
```diff
@@ -35,7 +35,10 @@ def main():
     # RealFakeDataseta will limit the number of images sampled per dataset to the length of the smallest dataset
     real_datasets, fake_datasets = load_datasets()
     train_dataset, val_dataset, test_dataset = create_real_fake_datasets(
-        real_datasets, fake_datasets, base_transforms, None) #random_image_transforms)
+        real_datasets,
+        fake_datasets,
+        base_transforms,
+        train_augmentations=random_aug_transforms)
```
Comment (aliang322) [inline base_miner/train_detector.py]: Should this be `train_transforms=random_aug_transforms`?

**Review by aliang322 — COMMENTED** [review]
Transform refactoring looks good. Added minor line comments

**Comment by benliang99** [comment]
Looks good, had the same question as Andrew about either dropping 4 channel images (dropping the alpha channel) or some other approach.

File: bitmind/image_transforms.py (hunk 1 above)
Comment (benliang99) [reply bitmind/image_transforms.py]: Looks good, had the same question as Andrew about either dropping 4 channel images (dropping the alpha channel) or some other approach.

File: base_miner/train_detector.py (hunk 4 above)
Comment (benliang99) [reply base_miner/train_detector.py]: Clarification: Is this missing a parameter?
From [util/data.py](https://github.com/BitMind-AI/bitmind-subnet/blob/0595530fbae2a627cb270ff4d99a9b869d777503/base_miner/util/data.py):
```
def create_real_fake_datasets(
    real_datasets: Dict[str, List[ImageDataset]],
    fake_datasets: Dict[str, List[ImageDataset]],
    train_transforms: transforms.Compose,
    val_transforms: transforms.Compose,
    test_transforms: transforms.Compose,
) -> Tuple[RealFakeDataset, ...]:
```

**Review by benliang99 — COMMENTED** [review]
Added some comments for clarification.

File: bitmind/image_transforms.py (hunk 3 above)
Comment (dylanuys) [reply bitmind/image_transforms.py]: Good point - I realize now that the ImageDataset class handles alpha channels (prior to calling any transforms), but validator transforms don't. Replaced this lambda with `.convert('RGB')` to handle both L and RGBA

File: bitmind/constants.py (hunk 2 above)
Comment (dylanuys) [reply bitmind/constants.py:45]: Yes, as well as to describe images accepted by the base miner model. This variable and the transforms are used on both sides by default for consistency.

**Review by aliang322 — APPROVED** [review]
Looks good. Comments were addressed by subsequent commits

**Pull request merged**